tests/checks/path_prefix/path_prefix_test.go

```diff
@@ -0,0 +1,222 @@
+`
+
+	httpScaledObjectTemplate = `
+kind: HTTPScaledObject
+apiVersion: http.keda.sh/v1alpha1
+metadata:
+  name: {{.HTTPScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  host: {{.Host}}
+  pathPrefixes:
+  - {{.PathPrefix0}}
+  - {{.PathPrefix1}}
+  targetPendingRequests: 100
+  scaledownPeriod: 10
+  scaleTargetRef:
+    deployment: {{.DeploymentName}}
+    service: {{.ServiceName}}
+    port: 8080
+  replicas:
+    min: {{ .MinReplicas }}
+    max: {{ .MaxReplicas }}
+`
+)
+
+func TestCheck(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 6, 10),
+		"replica count should be %d after 1 minute", minReplicaCount)
+
+	testScale(t, kc, data, templates, pathPrefix0)
+	testNotScale(t, kc, data, templates, notPathPrefix0)
+	testScale(t, kc, data, templates, pathPrefix1)
+	testNotScale(t, kc, data, templates, notPathPrefix1)
+
+	// cleanup
+	DeleteNamespace(t, testNamespace)
+}
+
+func testScale(t *testing.T, kc *kubernetes.Clientset, data templateData, templates []Template, pathPrefix string) {
+	t.Log("--- testing scale ---")
+
+	// create resources
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 6, 10),
+		"replica count should be %d after 1 minute", minReplicaCount)
+
+	data.PathPrefix = pathPrefix
+
+	// scale in
+	KubectlApplyWithTemplate(t, data, "loadJobTemplate", loadJobTemplate)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, maxReplicaCount, 6, 10),
+		"replica count should be %d after 1 minute", maxReplicaCount)
+
+	// scale out
+	KubectlDeleteWithTemplate(t, data, "loadJobTemplate", loadJobTemplate)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 12, 10),
+		"replica count should be %d after 2 minutes", minReplicaCount)
+
+	// cleanup
+	KubectlDeleteMultipleWithTemplate(t, data, templates)
+}
+
+func testNotScale(t *testing.T, kc *kubernetes.Clientset, data templateData, templates []Template, pathPrefix string) {
+	t.Log("--- testing not scale ---")
+
+	// create resources
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 6, 10),
+		"replica count should be %d after 1 minutes", minReplicaCount)
+
+	data.PathPrefix = pathPrefix
+
+	// not scale
+	KubectlApplyWithTemplate(t, data, "loadJobTemplate", loadJobTemplate)
+	assert.True(t, AssertDeploymentReplicaReadyCountForDuration(t, kc, deploymentName, testNamespace, minReplicaCount, 3*time.Minute),
+		"replica count should be %d for 3 minutes", minReplicaCount)
```

Comment:
I guess that one minute could be enough for checking that it doesn't scale and we don't increase too much the e2e test duration
```suggestion
	KubectlApplyWithTemplate(t, data, "loadJobTemplate", loadJobTemplate)
	assert.True(t, AssertDeploymentReplicaReadyCountForDuration(t, kc, deploymentName, testNamespace, minReplicaCount, 1*time.Minute),
		"replica count should be %d for 1 minutes", minReplicaCount)
```
Have you set 3 for any specific reason?

Reply:
1 minute is the timeout we define for the scale up operations on the other tests. If for some reason the deadline is too tight and there is a bug that misroutes the request, the deployment may be scaled but we did not observed it long enough to detect the problem. In that scenarios, all tests are successful but we have a problem.

I know that we would have to be very unlucky to meet all conditions necessary for this to happen, but I was trying to make the test more resilient.

What about we meet in the middle and set the value to 2 minutes?

Reply:
Agree :)